routing/base/astar_algorithm.hpp

```diff
@@ -349,6 +369,14 @@ typename AStarAlgorithm<TGraph>::Result AStarAlgorithm<TGraph>::FindPathBidirect
         if (!foundAnyPath || bestPathReducedLength > curPathReducedLength)
         {
           bestPathReducedLength = curPathReducedLength;
+          bestPathRealLength = stateV.distance + len - pV + distW;
+          bestPathRealLength += cur->pS;
+
+          if (distW > 0)
```

Comment:
Ok. distW do not best option for it. But I must to check if second wave is empty. We do not have heuristics in weight for this case.